nomos-mix/core/src/message_blend/mod.rs

```diff
@@ -0,0 +1,72 @@
+mod crypto;
+mod temporal;
+
+pub use crypto::CryptographicProcessorSettings;
+pub use temporal::TemporalProcessorSettings;
+
+use serde::{Deserialize, Serialize};
+use tokio::sync::mpsc;
+
+use crate::message_blend::crypto::CryptographicProcessor;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct MessageBlendSettings {
+    pub cryptographic_processor: CryptographicProcessorSettings,
+    pub temporal_processor: TemporalProcessorSettings,
+}
+
+/// [`MessageBlend`] handles the entire Tier-2 spec.
+/// - Wraps new messages using [`CryptographicProcessor`]
+/// - Unwraps incoming messages received from network using [`CryptographicProcessor`]
+/// - Pushes unwrapped messages to [`TemporalProcessor`]
+/// - Releases messages returned by [`TemporalProcessor`] to the proper channel
+pub struct MessageBlend {
+    /// To receive new messages originated from this node
+    new_message_receiver: mpsc::UnboundedReceiver<Vec<u8>>,
+    /// To receive incoming messages from the network
+    inbound_message_receiver: mpsc::UnboundedReceiver<Vec<u8>>,
+    /// To release messages that are successfully processed but still wrapped
+    outbound_message_sender: mpsc::UnboundedSender<Vec<u8>>,
+    /// To release fully unwrapped messages
+    fully_unwrapped_message_sender: mpsc::UnboundedSender<Vec<u8>>,
+    /// Processor
+    cryptographic_processor: CryptographicProcessor,
+    // TODO: Add TemporalProcessor
+}
```

Comment:
Nitpick: Probably the most elegant way of doing this would be to implement `Sink` and `Stream` instead of using channels. This is an idea that would have to be specified but I Imagine the distinct tiers as stream operation that could be mapped maybe 🤔 

Again, nothing incorrect with this, is straight forward. But if at some point you want to explore into rust black magic we could do it.

Reply:
True. Actually, I tried to implement `Sink` and `Stream` at the beginning, but turned into this way because of some small issues. But now, I think it's possible (as I did in https://github.com/logos-co/nomos-node/pull/846). Let me try in a new PR. There might be some small issues to discuss, but I agree that it's better to implement Sink and Stream instead of using channels in the library level.

Reply:
This refactoring will be done as a next PR. First of all, we're going to have a [discussion](https://github.com/logos-co/nomos-node/pull/847#pullrequestreview-2394511285) to make things simpler.